From 5c19a02979d3414f1326359494c1df38eac2730b Mon Sep 17 00:00:00 2001
From: Mingli Yu <mingli.yu@windriver.com>
Date: Wed, 11 May 2022 17:47:09 +0800
Subject: [PATCH] kni_net.c: use netif_rx() instead of netif_rx_ni()

The netif_rx_ni is removed in below commit after kernel upgraded
to 5.18, so change accordingly to fix the gap.
2655926aea9b net: Remove netif_rx_any_context() and netif_rx_ni().

Upstream-Status: Submitted [https://mails.dpdk.org/archives/dev/2022-May/240924.html]

Signed-off-by: Mingli Yu <mingli.yu@windriver.com>

---
 kernel/linux/kni/kni_net.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/kernel/linux/kni/kni_net.c b/kernel/linux/kni/kni_net.c
index 29e5b9e21f..e66b35314a 100644
--- a/kernel/linux/kni/kni_net.c
+++ b/kernel/linux/kni/kni_net.c
@@ -441,7 +441,7 @@ kni_net_rx_normal(struct kni_dev *kni)
 		skb->ip_summed = CHECKSUM_UNNECESSARY;
 
 		/* Call netif interface */
-		netif_rx_ni(skb);
+		netif_rx(skb);
 
 		/* Update statistics */
 		dev->stats.rx_bytes += len;
